### ISSUE TITLE
The ascii characters rendered by the external font are particularly wide. I don't know if there was something wrong.

### ISSUE THREAD

**Issue by vvlookman** [sharazam/printpdf#201]
<img width="767" alt="image" src="https://github.com/user-attachments/assets/e482e228-5b5e-4412-b598-582842d7212d" />

**Comment by vvlookman** [comment]
Even if you copy the built-in font file and load it as an external font, the rendering effect will be different from using the built-in font directly, and it will be also so wide

**Comment by fschutt** [comment]
Yeah, it means the widths of the glyphs are not parsed / encoded correctly. 

What font is this? Does your font support vertical writing (in that case it could be because it's using the `/DW2` instead of the `/DW` dictionary)?

printpdf usually normalizes character widths but falls back to the "default" width if the width of a glyph is not found, and that width is a square, i.e. for the glyph `width = height`, which is why they appear oddly spaced.

**Comment by vvlookman** [comment]
This should have nothing to do with fonts, because I have also tested built-in fonts, and directly imported the ttf file of the built-in fonts as an external font will have the same effect.

My workaround is to separate ASCII characters from non-ASCII characters in the text. ASCII characters are rendered using built-in fonts and non-ASCII characters are rendered using external fonts. So, this problem may be that there may be a problem with the way external fonts calculate character widths?

**Comment by vvlookman** [comment]
> Yeah, it means the widths of the glyphs are not parsed / encoded correctly.
> 
> What font is this? Does your font support vertical writing (in that case it could be because it's using the `/DW2` instead of the `/DW` dictionary)?
> 
> printpdf usually normalizes character widths but falls back to the "default" width if the width of a glyph is not found, and that width is a square, i.e. for the glyph `width = height`, which is why they appear oddly spaced.

The font is [Noto Sans CJK](https://github.com/googlefonts/noto-cjk/tree/main/Sans#downloading-noto-sans-cjk)

**Comment by fschutt** [comment]
Fixed in 0.8. Can be closed now. 

https://crates.io/crates/printpdf
